File: 137_Single_Number_II/single_number_ii.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        return ( sum(set(nums)) * 3 - sum(nums) ) / 2
        """
        ans = 0
        for i in xrange(32):
            sum = 0
            for n in nums:
                if n >> i & 1:
                    sum += 1
            sum %= 3
            ans |= (sum << i)
        if ans > 0x7FFFFFFF:
            return -int(0x100000000 - ans)
        return ans
        """

File: 137_Single_Number_II/single_number_ii.java
public class Solution {
    public int singleNumber(int[] nums) {
        Arrays.sort(nums);
        int prev = nums[0];
        int ref = 1;
        for (int i = 1; i < nums.length; ++i) {
            if (nums[i] != prev) {
                if (ref < 3) {
                    return prev;
                } else {
                    prev = nums[i];
                    ref = 1;
                }
            } else {
                ref += 1;
            }
        }
        return prev;
    }
}
